textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/fuzzytokenmatcher/FuzzyTokenMatcher.java

```diff
@@ -28,11 +28,17 @@
     private List<Attribute> attributeList;
     private int threshold;
     private ArrayList<String> queryTokens;
+    private int limit;
+    private int cursor;
+    private int offset;
 
     public FuzzyTokenMatcher(IPredicate predicate) {
-    	this.predicate = (FuzzyTokenPredicate)predicate;
-    	DataReaderPredicate dataReaderPredicate = this.predicate.getDataReaderPredicate();
-    	this.sourceOperator = new IndexBasedSourceOperator(dataReaderPredicate);
+        this.cursor = 0;
```

Comment:
Let's use "cursor" to point to the current record to process.  So let's use "-1" initially.